src/users/users.controller.js

```diff
@@ -0,0 +1,123 @@
+import {
+  Controller,
+  Dependencies,
+  UseGuards,
+  Get,
+  Post,
+  Put,
+  Delete,
+  Bind,
+  Body,
+  Param,
+  Query,
+  BadRequestException,
+  NotFoundException,
+} from '@nestjs/common';
+import { AuthGuard } from '@nestjs/passport';
+import { UsersService } from './users.service';
+
+@Controller('users')
+@Dependencies(UsersService)
+export class UsersController {
+  constructor(usersService) {
+    this.usersService = usersService;
+  }
+
+  @UseGuards(AuthGuard('HMAC'))
+  @Post('')
```

Comment:
Why do you prefer `POST` to `PUT` for user creation?

Reply:
There are some business considerations.

Generally a user id may not uniquely identify a person in the real world if the consumer don't assign special meaning to it (e.g., passport number, employee id), so I think the consumer don't want to replace an existing user with same id, and `POST` is preferred.

If the consumer system use unique identifier of a person in the real world (e.g., passport number), then I think `PUT` (replace) may be better in this scenario.